### ISSUE TITLE
Implement dataclass decorator?

### ISSUE THREAD

**Issue by jlandmann** [GLAMOS/dataflow#5]
Just another suggestion: It's a Python 3.7 feature, but might be handy e.g. for the MassBalance class
See [PEP 557](https://www.python.org/dev/peps/pep-0557/)

**Comment by yweidmann** [comment]
Good point. As soon as we migrate to Python 3.7 we should add the dataclass decorator to all classes of the package DataObjects (except of the Enumerations and Exceptions).
